### PR TITLE
Feat : Create a basic contract for proof api health check

### PR DESCRIPTION
**Pull request by OBlackmon3**
## What?
Create a smart contract to generate events for Proof API
## Why?
We need a consistent source of events to prove on other chains
## How?
The contract has a method that accepts transactions
## Testing?
N/A

## Summary by CodeRabbit

- **New Features**
	- Introduced a smart contract for managing Ether transactions and logging events.
	- Added functionality for receiving Ether and sending it to specified recipients.
	- Implemented event logging for transaction details, including health checks and transfers.
	- Provided a method to retrieve the current Ether balance of the contract.

### CODE REVIEW

File: contracts/utils/EventCreation.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract SimpleTransaction {
```
Comment (RnkSngh) [inline contracts/utils/EventCreation.sol:4]: nit: like i don't have strong opinions on this or anything but we did kinda have a planet naming thing going on for example contracts

but I can't emphasize enough. I don't have any strong opinions on this. 

But we did have a planet thing going on

File: contracts/utils/EventCreation.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    address public owner;
```
Comment (RnkSngh) [inline contracts/utils/EventCreation.sol:5]: consider inheriting from [openzeppelin's Ownable](https://docs.openzeppelin.com/contracts/2.x/access-control#ownership-and-ownable) & use `onlyOwner` modifier for sending

**Review by RnkSngh — APPROVED** [review]

**Comment by RnkSngh** [comment]
might be nice if we had some sort of dynamic event emitting. Wondereing if we can do `emit 0x...` with arbitrary bytes or if compiler won't like that

File: contracts/utils/ProofAPIEventGenerator.sol
Comment (RnkSngh) [inline contracts/utils/ProofAPIEventGenerator.sol:1]: hmm let's put it in the same directory as the other example contracts like mars & venus 

`contracts/examples/`

File: contracts/utils/ProofAPIEventGenerator.sol [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    // Event to log the transaction
+    event ProofAPIEventHealthCheck(address indexed to, uint256 amount, uint256 srcChainID, uint256 destChainID);
+
+    // Constructor to set the contract owner
+    constructor() {
+        owner = msg.sender;
+    }
+
+    // Function to receive Ether
+    receive(uint256 amount uint256 srcChainID, uint256 destChainID) external payable {
```
Comment (RnkSngh) [inline contracts/utils/ProofAPIEventGenerator.sol:16]: make sure code compiles using `forge build`
```suggestion
    receive(uint256 amount, uint256 srcChainID, uint256 destChainID) external payable {
```

**Review by RnkSngh — APPROVED** [review]

File: contracts/utils/EventCreation.sol (hunk 2 above)
Comment (OBlackmon3) [reply contracts/utils/EventCreation.sol]: Will do!

File: contracts/utils/EventCreation.sol (hunk 1 above)
Comment (OBlackmon3) [reply contracts/utils/EventCreation.sol]: Thank you. Will rename to mercury

**Comment by OBlackmon3** [comment]
No longer needed for now

**Pull request closed without merging**